part2.md

```diff
@@ -0,0 +1,129 @@
+## Part 2 - Sprinkle some types
+
+ * Make sure that Typescript is installed:
+ ```
+ npm install -g typescript
+ ```
+ 
+ * Make sure that Yarn is installed:
+ ```
+ npm install -g yarn
+ ```
+ 
+ * Add configuration to `package.json` (note dependencies and scripts):
+ ``` js
+ {
+  "name": "functions",
+  "description": "Cloud Functions for Firebase",
+  "dependencies": {
+    "firebase-admin": "~5.4.0",
+    "firebase-functions": "^0.7.0"
+  },
+  "devDependencies": {
+    "typescript": "^2.3.2"
+  },
+  "scripts": {
+    "build": "tsc",
+    "deploy": "tsc && firebase deploy --only functions"
+  },
+  "main": "build/index.js",
+  "private": true
+}
+  ```
+
+ * Add `tsconfig.json` next to `package.json`:
+``` js
+ {
+    "compilerOptions": {
+        "lib": [
+            "es6",
+            "es2015.promise"
+        ],
+        "module": "commonjs",
+        "noImplicitAny": false,
+        "outDir": "build",
+        "sourceMap": true,
+        "target": "es6"
+    },
+    "include": [
+        "src/**/*.ts"
+    ]
+}
+```
+
+ * Create `src/index.ts` file with firebase imports:
+``` js
+import * as functions from 'firebase-functions'
+import * as admin from 'firebase-admin'
+
+admin.initializeApp(functions.config().firebase)
+```
+
+ * Add basic class definitions for tables in database: 
+``` ts
+class Rates {
+    usd: number
+    czk: number
+    pln: number
+    rub: number
+}
+
+class Conversions extends Rates {
```

Comment:
Is this a single or multiple Conversion? why is the class name plural?

Reply:
"conversions" to different currencies.